test/NuGet.Clients.Tests/NuGet.VisualStudio.Common.Test/NuGetExperimentationServiceTests.cs

```diff
@@ -3,121 +3,153 @@
 
 using System;
 using System.Collections.Generic;
-using System.Threading;
-using System.Threading.Tasks;
-using System.Windows.Media;
 using FluentAssertions;
 using Microsoft.VisualStudio.Experimentation;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Moq;
 using NuGet.Common;
 using Test.Utility;
+using Test.Utility.VisualStudio;
 using Xunit;
 
 namespace NuGet.VisualStudio.Common.Test
 {
-    [Collection(nameof(TestJoinableTaskFactoryCollection))]
+    [Collection(MockedVS.Collection)]
     public class NuGetExperimentationServiceTests
     {
+        private readonly Mock<IOutputConsoleProvider> _outputConsoleProviderMock;
         private readonly Lazy<IOutputConsoleProvider> _outputConsoleProvider;
+        private readonly Mock<IOutputConsole> _outputConsoleMock;
 
-        public NuGetExperimentationServiceTests()
+        public NuGetExperimentationServiceTests(GlobalServiceProvider sp)
         {
-            _outputConsoleProvider = new Lazy<IOutputConsoleProvider>(() => new TestOutputConsoleProvider());
-        }
+            sp.Reset();
 
-        private IList<string> OutputMessages => ((TestOutputConsoleProvider)_outputConsoleProvider.Value).TestOutputConsole.Messages;
+            var mockOutputConsoleUtility = OutputConsoleUtility.GetMock();
+            _outputConsoleProviderMock = mockOutputConsoleUtility.mockIOutputConsoleProvider;
+            _outputConsoleProvider = new Lazy<IOutputConsoleProvider>(() => _outputConsoleProviderMock.Object);
+            _outputConsoleMock = mockOutputConsoleUtility.mockIOutputConsole;
+        }
 
         [Fact]
-        public void Constructor_WithNullWrapper_Throws()
+        public void Constructor_WithNullEnvironmentVariableReader_Throws()
         {
-            Assert.Throws<ArgumentNullException>(() => new NuGetExperimentationService(null, new TestVisualStudioExperimentalService(), _outputConsoleProvider));
+            Assert.Throws<ArgumentNullException>(() => new NuGetExperimentationService(environmentVariableReader: null, Mock.Of<IExperimentationService>(), Mock.Of<Lazy<IOutputConsoleProvider>>()));
         }
 
         [Fact]
         public void Constructor_WithNullExperimentalService_Throws()
         {
-            Assert.Throws<ArgumentNullException>(() => new NuGetExperimentationService(new TestEnvironmentVariableReader(new Dictionary<string, string>()), null, _outputConsoleProvider));
+            Assert.Throws<ArgumentNullException>(() => new NuGetExperimentationService(Mock.Of< IEnvironmentVariableReader>(), experimentationService: null, _outputConsoleProvider));
```

Comment:
```suggestion
            Assert.Throws<ArgumentNullException>(() => new NuGetExperimentationService(Mock.Of<IEnvironmentVariableReader>(), experimentationService: null, _outputConsoleProvider));
```

Reply:
Good 👀s. I'd like to get this merged, so will fix that in my next PR.